File: apps/yakoon-app-console/src/yakoon/app_console/renderer/rule.py
from .base import BaseRenderer


class RuleRenderer(BaseRenderer):

    def __init__(self, node):
        self.node = node

    def render(self) -> str:
        style = self.node.props.get("style", "normal")

        if style == "strong":
            return "━━━━━━━━━━━━━━━━━━━━━━━━\n"
        if style == "subtle":
            return "------------------------\n"

        return "────────────────────────\n"

File: clients/yakoon-client-web/js/renderer/block/rule.js
import { registerBlock } from "./core.js";
import { createElement } from "../../dom.js";

registerBlock("rule", (node) => {
    const el = createElement("hr");

    el.classList.add("rule");

    const style = node.props.style || "normal";
    el.classList.add(`rule-${style}`);

    return el;
});